server/src/domains/auth/auth.service.ts

```diff
@@ -0,0 +1,146 @@
+import { HttpException, HttpStatus, Injectable, UnauthorizedException } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { UserPayload } from './dto/userPayload.dto';
+import { Token } from './entities/token.entity';
+import * as bcrypt from 'bcryptjs'
+import { CreateUserDto } from '../users/dto/create-user.dto';
+import { UsersService } from '../users/users.service';
+import { SignInDto } from './dto/signin.dto';
+import { TokensReturnDto } from './dto/tokensReturn.dto';
+import { AuthReturnDto } from './dto/authReturn.dto';
+
+@Injectable()
+export class AuthService {
+    constructor(
+        @InjectRepository(Token)
+        private tokensRepository: Repository<Token>,
+        private jwtService: JwtService,
+        private usersService: UsersService
+    ) { }
+
+    generateTokens(payload: UserPayload): TokensReturnDto {
+        const accessToken = this.jwtService.sign(payload, { expiresIn: '30m' })
+        const refreshToken = this.jwtService.sign(payload, { expiresIn: '2d' })
+
+        return { accessToken, refreshToken }
+    }
+
+    validateToken(token: string): UserPayload {
```

Comment:
Since this method is returning user payload I propose to chance its declaration to
```typescript
    getUserPayloadFromToken(token: string): UserPayload | null {
```
Notice the null type that shows that the value is nullable and makes it easier to use it.